### PR TITLE
[GEOT-7112] Introduce DefaultLoggingFactory and revise logging integration tests to fail build

### PR DESCRIPTION
**Pull request by jodygarnett**
[GEOT-7112](https://osgeo-org.atlassian.net/browse/GEOT-7112) [Powered by Pull Request Badge](https://pullrequestbadge.com/?utm_medium=github&utm_source=geotools&utm_campaign=badge_info)

I was missing the addition of DefaultLoggingFactory from https://github.com/geotools/geotools/pull/3809 which makes GeoTools.init() process more understandable.

This pull-request also notes that logging integration tests were not being compiled, and thus not in position to run :)
  
# Checklist

- [x] I have read the [contribution guidelines](https://github.com/geotools/geotools/blob/main/CONTRIBUTING.md).
- [x] I have sent a [Contribution Licence Agreement](https://docs.geotools.org/latest/developer/procedures/contribution_license.html) (not required for small changes, e.g., fixing typos in documentation).
- [x] First PR targets the `main` branch (backports managed later; ignore for branch specific issues).
- [ ] Avoid [Java 9+ split packages](http://tutorials.jenkov.com/java/modules.html#split-packages-not-allowed).
- [x] All the build checks are green ([see automated QA checks](https://docs.geotools.org/latest/developer/conventions/code/qa.html)).

For core and extension modules:

- [x] New unit tests have been added covering the changes.
- [x] [Documentation](https://github.com/geotools/geotools/tree/main/docs) has been updated (if change is visible to end users).
- [x] There is an issue in [GeoTools Jira](https://osgeo-org.atlassian.net/projects/GEOT) (except for changes not visible to end users). 
- [ ] Commit message(s) must be in the form ``[GEOT-XYZW] Title of the Jira ticket``.
- [ ] Bug fixes and small new features are presented as a single commit.
- [ ] The commit targets a single objective (if multiple focuses cannot be avoided, each one is in its own commit, and has a separate ticket describing it).

### CODE REVIEW

**Comment by jodygarnett** [comment]
This is primarily a documentation change I would love to ask for a review form @ianturton or @bradh :)
Please be a little kind I am cleaning up some very old (and incorrect) docs from a long time ago...

File: docs/user/library/metadata/logging/log4j.rst [hunk 1]
```diff
@@ -70,33 +79,108 @@ All logging frameworks have some mechanism to delegate to java util logging as a
   FINEST        FINEST
   ALL           ALL
   ============= ================
+
+
+* Lots of ways to enable java util logging bridge to Log4J:
```
Comment (bradh) [inline docs/user/library/metadata/logging/log4j.rst:84]:
```suggestion
* There are several ways to enable ``java.util.logging`` bridge to Log4J:
```

File: docs/user/library/metadata/logging/logback.rst [hunk 2]
```diff
@@ -181,4 +157,47 @@ In a more complicated setup using multiple libraries you may also end up includi
...
+'''''''''''''''''''
+
+The following example is taken from our integration testing, this test *only* has slf4j api available so ``GeoTools.init()` is able to unambiguously determine ``LogbackLoggerFactory`` can be used.
+
+1. Setup :file:`pom.xml` with dependencies on geotools and Logback:
+
+   .. literalinclude:: /../../modules/library/metadata/src/it/logback/pom.xml
+      :language: xml
+      
+2. Configure logback wtih :download:`logback.xml </../../modules/library/metadata/src/it/logback/src/main/resources/logback.xml>` added to :file:`src/main/resources`:
```
Comment (bradh) [inline docs/user/library/metadata/logging/logback.rst:172]:
```suggestion
2. Configure logback with :download:`logback.xml </../../modules/library/metadata/src/it/logback/src/main/resources/logback.xml>` added to :file:`src/main/resources`:
```

**Comment by jodygarnett** [comment]
> Made a few doc comments. Didn't really look much at the code though.

Perfect, thanks @bradh !

**Pull request merged**